### PR TITLE
@itwin/core-react deprecations

### PR DESCRIPTION
**Pull request by GerardasB**
## Changes

This PR deprecates `@itwin/core-react` components that are replaced by iTwinUI equivalents.

## Testing

N/A

### CODE REVIEW

**Review by saskliutas — APPROVED** [review]

File: ui/core-react/src/core-react/inputs/LabeledComponentProps.tsx [hunk 1]
```diff
@@ -7,15 +7,18 @@
  */
 
 import type * as React from "react";
+import type { InputLabel } from "./InputLabel";
```
Comment (ignas-k) [inline ui/core-react/src/core-react/inputs/LabeledComponentProps.tsx:10]: Forgot to remove?

File: ui/core-react/src/core-react/button/UnderlinedButton.tsx [hunk 2]
```diff
@@ -13,6 +13,7 @@ import { Key } from "ts-key-enum";
 
 /** Properties for the [[UnderlinedButton]] React component
  * @public
+ * @deprecated in 4.12.x. Props of deprecated component {@link UnderlinedButton}.
```
Comment (ignas-k) [inline ui/core-react/src/core-react/button/UnderlinedButton.tsx:16]: Nitpick: should be `4.13.x` now. Applies to all deprecations

**Review by ignas-k — APPROVED** [review]
Awesome, just left some nitpicks,

File: ui/core-react/src/core-react/inputs/InputStatus.ts [hunk 3]
```diff
@@ -6,8 +6,11 @@
  * @module Inputs
  */
 
+import type { InputLabel } from "./InputLabel";
```
[the remark this answers is not in the document]
Comment (GerardasB) [reply ui/core-react/src/core-react/inputs/InputStatus.ts:9]: It's required for `@link` tag.

File: ui/core-react/src/core-react/inputs/LabeledComponentProps.tsx (hunk 1 above)
Comment (GerardasB) [reply ui/core-react/src/core-react/inputs/LabeledComponentProps.tsx:10]: It's required for `@link` tag.

**Pull request merged**